### ISSUE TITLE
How do you troubleshoot this? (I can't get tab behavior to work)

### ISSUE THREAD

**Issue by NullVoxPopuli** [abecodes/tabout.nvim#23]
I have nvim-treesitter (and this plugin, via packer)), but when I press tab near a ( or {, nothing happens. 

How do I troubleshoot what could be going wrong?

**Comment by abecodes** [comment]
Hi @NullVoxPopuli 

You could clone the repo, change the path in packer to the local folder and change [this line](https://github.com/abecodes/tabout.nvim/blob/6ff556b1b2274c8ba3eaedbf62339789e79baca8/lua/tabout/config.lua#L25) to `true`.

Let me know if you can identify the issue

**Comment by NullVoxPopuli** [comment]
what does that do? does it log somewhere?

**Comment by NullVoxPopuli** [comment]
Should `:TaboutToggle` be an editor command? I'm having doubts about.. _everything_ :upside_down_face:

**Comment by abecodes** [comment]
You should not need to use editor commands. If you follow the readme `tabout` should work out of the box. 

That means, that if you are inside brackets or similar symbols in a recognized node and using the `tab` key (insert mode), a tabout will be triggered and the cursor will jump behind the closing symbol. 

If you enable the debug mode, the node you are in will be logged to the message line of nvim.

Basically it should just work out of the box. Maybe you have an example scenario where it fails to work?

**Comment by NullVoxPopuli** [comment]
> If you enable the debug mode, the node you are in will be logged to the message line of nvim.

hm, yeah, I saw none of that

> Maybe you have an example scenario where it fails to work?

I do, in my dottfiles here: https://github.com/NullVoxPopuli/dotfiles/blob/master/home/.config/nvim/init.vim
here is the place where I setup tabout: https://github.com/NullVoxPopuli/dotfiles/blob/master/home/.config/nvim/lua/plugins/init.lua#L50 (which, from what I gather from the README, is all I need?)

**Comment by abecodes** [comment]
The good news: It works with your init file (except for the comment, but thats expected, since it is different node)

I guess you are just missing the call to the setup function, try smth like this:

```
use {
  'abecodes/tabout.nvim',
  config = function()
    require('tabout').setup {}
  end,
	wants = {'nvim-treesitter'}, -- or require if not used so far
	after = {'completion-nvim'} -- if a completion plugin is using tabs load it before
}
```

**Comment by NullVoxPopuli** [comment]
😢 I don't understand

I now have this:
![image](https://user-images.githubusercontent.com/199018/156885032-35ea0a2b-bbde-4746-ac72-c8547160fc26.png)

and run `:PackerSync`
and then close nvim
then open nvim in a new terminal
and when I press tab here:
![image](https://user-images.githubusercontent.com/199018/156885046-9a0e27b4-52aa-4337-b7cb-b28aa4cc0e90.png)

nothing happens.

Is there a way to see what all is handling or has ever registered against the tab key?

**Comment by abecodes** [comment]
You can type `:imap <TAB>`. Then you should only see one line like `i  <Tab>         !pumvisible() ? "<Plug>(Tabout)" : pumvisible() ? "<C-N>" : "<Plug>Tabout" `.

**Comment by NullVoxPopuli** [comment]
excellent, that is what I see.
![image](https://user-images.githubusercontent.com/199018/156892246-a1694a58-99d6-46e3-ba3a-83645678a9bb.png)

I think maybe I misunderstood what the intent is supposed to be, I've made a discovery

so, for example:
`'|some string'` where the character is at the beginning of the string, the tab key works
but if I try the same on a paren, 
```
|('some string'

// closing is far away
)
```
it doesn't work -- so as long as I understand this is intended for when the matching/closing character is on the same line, I think everything works 🤷 🙃

**Comment by abecodes** [comment]
You can use a different keybinding for `tabout` over multiple lines 😉 

```
vim.api.nvim_set_keymap('i', '<A-x>', "<Plug>(TaboutMulti)", {silent = true})
vim.api.nvim_set_keymap('i', '<A-z>', "<Plug>(TaboutBackMulti)", {silent = true})
```
